api/include/opentelemetry/http/http_client.h

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft. All rights reserved
+
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace http
+{
+
+
+enum class HttpResult: uint8_t
+{
+    // Response has been received successfully from target server.
+    HttpResult_OK = 0,
+
+    // Request has been aborted by the caller. The server might or
+    // might not have already received or processed the request.
+    HttpResult_Aborted = 1,
+
+    // Local conditions have prevented the request from being sent
+    // (invalid request parameters, out of memory, internal error etc.)
+    HttpResult_LocalFailure = 2,
+
+    // Network conditions somewhere between the local machine and
+    // the target server have caused the request to fail
+    // (connection failed, connection dropped abruptly etc.).
+    HttpResult_NetworkFailure = 3
+
+};
+
+// The HttpHeaders class contains a set of HTTP headers.
+class HttpHeaders
+{
+public:
+    // Inserts a name/value pair, and removes elements with the same name.
+    virtual void set(nostd::string_view const& name, nostd::string_view const& value) = 0;
+
+    // Inserts a name/value pair
+    virtual void add(nostd::string_view const& name, nostd::string_view const& value) = 0;
+
+    // Gets a string value given a name
+    virtual nostd::string_view const& get(nostd::string_view const& name) const = 0;
```

Comment:
Need to clarify what happens if there are multiple headers with the same name.

Reply:
Good point. Have clarified in comments.